src/lib/ngx-masonry.directive.ts

```diff
@@ -33,40 +35,57 @@ export class NgxMasonryDirective implements OnInit, OnDestroy, AfterViewInit {
     if (this.parent.options.animations !== undefined) {
       this.animations = this.parent.options.animations;
     }
+    if (this.parent.options.detectImageLoad === undefined) {
+      this.parent.options.detectImageLoad = this.detectImageLoad;
+    }
     this.renderer.setStyle(this.element.nativeElement, 'position', 'fixed');
     this.renderer.setStyle(this.element.nativeElement, 'right', '-150vw');
     this.parent.addPendingItem(this);
   }
 
   ngAfterViewInit() {
-    const images: HTMLImageElement[] = Array.from(this.element.nativeElement.getElementsByTagName('img'));
-    if (images.length === 0) {
+    // image lazyload or non-image support by selecting `itemSelector`
+    // image support by default
+    this.selectedElements = !this.parent.options.detectImageLoad
+      ? (this.selectedElements = this.element.nativeElement.innerHTML)
+      : Array.from(this.element.nativeElement.getElementsByTagName('img'));
+
+    if (this.selectedElements.length === 0) {
       setTimeout(() => {
         this.parent.add(this);
       });
     } else {
-      this.images = new Set(images);
-      for (const imageRef of images) {
-        this.renderer.listen(imageRef, 'load', _ => {
```

Comment:
Instead of having an option, could you use `nativeElement.getAttribute('loading')` here.

Reply:
change `this.renderer.listen(imageRef, 'load'` to `nativeElement.getAttribute('loading')`?

I don't think that would work if some image lazyload methods are not using `loading` attribute.

Reply:
Something like this
```
    if (imageRef.nativeElement.getAttribute('loading') == 'lazy') {
        this.imageLoaded(imageRef);
    } else {
        this.renderer.listen(imageRef, 'load', _ => {
          this.imageLoaded(imageRef);
        });
        this.renderer.listen(imageRef, 'error', _ => {
          this.imageLoaded(imageRef);
        });
    }
```
If you are using some more complicated methods like a scroll/viewport check, maybe you can add an `ngif` to the image tag?

Reply:
Can we still have a new option `lazyloadAttribute` for that `getAttribute(lazyloadAttribute)`?

`loading` only works for the native Chrome image lazyloading method.

Reply:
How about using a different attribute? Just add "masonryLazy", for example, to your image tag, and the library will ignore all images with this label.

Reply:
you meant?

Then how do we update on the readme? 
If `<img src='...'  masonryLazy>`, then this lib would not check if the image is loaded or not?

Also, we still need to add fixed width and height to all the div or image to prevent overlapping issue?

```
if (imageRef.nativeElement.getAttribute('masonryLazy') {
        this.imageLoaded(imageRef);
    } else {
        this.renderer.listen(imageRef, 'load', _ => {
          this.imageLoaded(imageRef);
        });
        this.renderer.listen(imageRef, 'error', _ => {
          this.imageLoaded(imageRef);
        });
    }
```

Reply:
Yeah, that's what I mean.
You must have fixed width and height before the image is loaded, otherwise there's no way for the lib to know the size of the item.

Reply:
ok. that should work as well. I will update the PR later today.

Reply:
Sounds good. Thank you